src/components/main-form.js

```diff
@@ -94,13 +93,90 @@ const InputWrapper = styled('div')`
   }
 `
 
-const ColHeader = styled(Col)`
-  font-weight: bold;
+const AmortizationTable = styled(Table)`
+  margin-top: 30px;
+
+  th,
+  tbody {
+    text-align: center;
+  }
+
+  th:first-of-type {
+    border: 0;
+  }
+
+  th:nth-child(6),
+  td:nth-child(6) {
+    border-left-width: thick;
+    border-left-color: ${DARK_GRAY}};
+  }
 `
 
+const centered = css`
+  margin: 0 auto;
+  display: block;
+`
```

Comment:
Ah, then you'd want https://getbootstrap.com/docs/4.3/utilities/display/

Reply:
so basically adding one class to make it a block, and another to center it? I think that having our own class `centered` that we can use around the app is a lot more readable than having 2 inline classes `d-block mx-auto`. It's pretty unlikely that I'll remember what `mx-auto` does, but I know for sure what `centered` does.

Reply:
`mx-auto` is part of the library, and it follows its standard. `ma(argin)x-auto`.

Here, we duplicated the library implementation and need to reference it outside of the place its used (which is once). If we are using this >3 times, I think it's worth extracting as a helper

Reply:
but `mx-auto` doesn't center it alone. You need to add a separate class to turn it into a block element as well `d-block` don't you? Assuming we will be using this >3 times since centering is something that we do often, how is having multiple unintuitive `mx-auto d-block`  classnames all over the place better than having a shorter and more intuitive `centered`  global class that we can control?

Reply:
> Assuming we will be using this >3 times since centering is something that we do often, how is having multiple unintuitive mx-auto d-block classnames all over the place better than having a shorter and more intuitive centered global class that we can control?

😂 I'm going to challenge you on this assumption on the grounds of #premature-optimization

Reply:
I get how you'd see this as premature optimization but 

```
  margin: 0 auto;
  display: block;
```
is one of the most popular ways to center elements. I can perfectly see us reusing this class and composing it into others. I get what you're saying because right now we only have 1 instance of the class, and there's definitely a chance that we'll build on top of this and get rid of it because we want to change how things look, but then again, I don't see the benefits of adding 2 less intuitive inline classes to save us a variable declaration which contains 2 more intuitive CSS properties. If at least the inline classes saved us from writing a huge block of CSS code I'd understand, but that's not the case.

Call it premature optimization but it's still a potential benefit. Where's the benefit with having `d-block mx-auto` instead?

Reply:
To remove the cost of looking up `centered` in the current work

We can also do `style="margin:0 auto;display:block;"` but this leads into another trap of high specificity and duplicate implementation to what Bootstrap already provides

Reply:
just to clarify, I'm not arguing about this cause I feel strongly about having it my way in the app, but more so to understand you so that I can be a better developer and understand other design patterns and their benefits

Reply:
Something to keep in mind is that I already approved this PR, and I'm happy to keep discussing this, though there isn't always one right solution.

Each implementation as its tradeoffs. I'm advocating that inlining this has slightly more benefits, but on the grander scheme of things, this piece of implementation is small.

Ultimately a goal we have as developers is to reduce complexity. We are discussing this to think about future patterns. For this particular case, I feel it to be a premature optimization that requires me to do more lookup. This is also b/c I have a good sense of what Boostrap and other design frameworks provide, so the reading comes natural. There is a learning curve to learn the syntax, but b/c I already paid that cost, I don't factor it into the calculation like you rightfully do

Reply:
Another thing to note is that design frameworks are built for use by large teams. This is why we may not want to write `margin: 0px` in a bunch of places separately, and instead want to write `centered`.

The language a team shares is given by a design framework so that code can be easily understood from many different places. That also ties into what I mean by the cost of reading a particular design syntax

Reply:
> To remove the cost of looking up `centered` in the current work

It took me a total of 2-3 minutes to look up `d-block` and `mx-auto` and for that I had to minimize my editor, use the internet, and scroll through documentation. At least I didn't have to look over any source code but if I ever have issues with those classes, I know I will. If I just take a quick look, I can see that `d-block` translates to `display: block !important` which isn't exactly what I want and could cause problems in the future. 

Since the classes are not as intuitive, it is likely that I'll repeat this process. I'll forget what they mean if I don't use them every day.  If I forget what `centered` means, which is less likely to happen, at least I won't go through all of the steps above. A small search will take me less than 30 seconds and won't require the use of an internet connection.

Reply:
> reading comes natural

at the expense of forgetting CSS! 😂

Reply:
took care of this and added `d-block mx-auto`. Please merge the branch after you see this comment!

Reply:
I guess we never established the norm, but I'm used to the writer merging and deploying their PR. The reviewer only approves a PR

Reply:
oh ok. I've done it both ways. If a developer asks me to merge their PR after it's been approved, I'll just do it 🤷‍♂